test/TestProjects/ExactMatchInheritance/Generated/Models/AzureResourceFlattenModelData.cs

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+namespace ExactMatchInheritance
+{
+    /// <summary> A class representing the AzureResourceFlattenModel data model. </summary>
+    public partial class AzureResourceFlattenModelData
```

Comment:
This is not correct. Probably I didn't compose a correct test data.

Reply:
Updated according to review comments. Now the correct type will be deduced. Thanks @m-nash !